src/test/coinscachepair_tests.cpp

```diff
@@ -154,16 +154,16 @@ BOOST_AUTO_TEST_CASE(linked_list_set_state)
     BOOST_CHECK_EQUAL(sentinel.second.Next(), &n1);
     BOOST_CHECK_EQUAL(sentinel.second.Prev(), &n1);
 
-    // Check that setting FRESH on new node inserts it after n1
-    CCoinsCacheEntry::SetFresh(n2, sentinel);
-    BOOST_CHECK(n2.second.IsFresh() && !n2.second.IsDirty());
+    // Check that setting DIRTY and FRESH on new node inserts it after n1
+    CCoinsCacheEntry::SetDirty(n2, sentinel, /*fresh=*/true);
+    BOOST_CHECK(n2.second.IsFresh() && n2.second.IsDirty());
```

Comment:
nit: for consistency with other `SetDirty` calls we could swap the parameter order:
```suggestion
    BOOST_CHECK(n2.second.IsDirty() && n2.second.IsFresh());
```

----

Alternatively we could rename `IsFresh` to `IsDirtyAndFresh` and assert that if fresh is true, so is dirty - which would simplify this line to asserting just a single method.